training/Exercise03_Forward_Models.ipynb

```diff
@@ -565,7 +565,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Some LinearModel objects simply contain a matrix representing the linear mapping; for those this matrix can be accessed by:"
+    "Some LinearModel objects simply contain a matrix representing the linear mapping; while others use a matrix-free approach. In both cases, one can extract the matrix of the linear model by (this will raise an error if the model is too large):"
```

Comment:
This is probably to keep in mind but not part of this PR, it could be a good idea to inform the user that a matrix being built when this is the case. i.e. when asking for matrix and the linear model is matrix-free. Do we do that?

Reply:
We don't at this point. Unsure if we want to.

Reply:
I would say if it is cached then probably it might be okay not to inform the user, but if it is not cached and computed every time the user asks for it then it will be helpful to update the user.

Reply:
Just checked, and we do cache it :)